h2o-core/src/main/java/water/parser/SyntheticColumnGenerator.java

```diff
@@ -17,6 +16,14 @@ public Frame finalize(Job<Frame> job, Vec[] parsedVecs, ParseSetup setup, int[]
       withSynth[parsedVecs.length + synthIdx] = parsedVecs[0].makeCon(Vec.T_STR);
     }
     new SyntheticColumnGeneratorTask(setup, fileChunkOffsets).doAll(withSynth);
+
+    if (Vec.T_CAT == setup._synthetic_column_type) {
```

Comment:
Currently, the `_synthetic_column_type` is not exposed to the user - it's only there to control the resulting vec type internally. Therefore it's not an array of types, but one general type of columns, because that's what is required right now. Could be changed in future if required.